myproject/leave/views.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from django.shortcuts import render
+
+# Create your views here.
+from rest_framework.views import APIView
+from models import Designation, User, Status, LeaveType, LeaveCredit, LeaveRequest
+
+from serializers import UserSerializer, LeaveRequestSerializer, LeaveCreditSerializer, StatusSerializer
+
+from serializers import UserSerializer, LeaveCreditSerializer, StatusSerializer, LeaveRequestSerializer2
+
+from rest_framework.response import Response
+from django.shortcuts import get_object_or_404
+from rest_framework.decorators import api_view
+from rest_framework.renderers import JSONRenderer, TemplateHTMLRenderer
+import json
+# import requests
+
+
+
+class Post(APIView):
+
+    def post(self, request):    
+        # response = requests.get(request.data)
+        # json_data = json.loads(response.text)
+        #value = request.data.encode("utf-8")
+        name = request.data["name"]
+
+        reporting_senior = request.data["reporting_senior"]
+        credits = request.data["credits"]
+        fromDate = request.data["fromDate"]
+        toDate = request.data["toDate"]
+        reason = request.data["reason"]
+        days = request.data["days"]
+        status = request.data["status"]
+        leave_request = LeaveRequest.objects.create(
+            employee_name= User.objects.get(name=name),
+            reporter = User.objects.get(name=reporting_senior), 
+            leave_type = LeaveType.objects.get(id=credits),
+            from_date = fromDate,
+            to_date = toDate,
+            no_days = days ,
+            reason = reason,
+            status = Status.objects.get(id=status)
+            )
+            
+        leave_request.save()
+
+        values = json.loads(request.body)
```

Comment:
What this method does. Saving a leave request a then returning a user?
Use meaning full names for methods?